src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,37 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
+            int length = Interop.GlobalizationNative.ToAscii(flags, unicode, count, null, 0);
```

Comment:
Right - so I don't get confused and do the wrong thing  are you suggesting we:
- change the formula for allocating the chars to 2*count+4 (may not work so well with domain labels or ascii)
- move the two interop calls to the non-stackalloced method (increases code complexity)
- keep the implementation as is currently in the PR (may not be optimised for all cases)